### PR TITLE
Singlefilecollab

### PR DESCRIPTION
**Pull request by dmaynard**
oxResourceIterSingle file collaborations.   Also fixed bug in BocResourceIterable which was causing the new style paging Iterators to loop indefinitely.

### CODE REVIEW

**Comment by boxcla** [comment]
Verified that @dmaynard has signed the CLA. Thanks for the pull request!

**Comment by dmaynard** [comment]
DO NOT merge yet.  I am still investigation a paging bug

**Comment by dmaynard** [comment]
OK. Please Review.  
Fixed the singleFileCollabSucceeds Integration Test.  Also fixed the bug where turning on logging caused Folder Rename to fail. It was another case of a response.disconnect() happening too soon

File: src/test/java/com/box/sdk/BoxCollaborationTest.java [hunk 1]
```diff
@@ -50,4 +67,71 @@ public void deleteSucceeds() {
...
+        }
+
+        assertEquals(collabIterator.hasNext(), false);
+        assertEquals(2, numCollabs);
+
+        uploadedFile.delete();
+
+
+    }
+
```
Comment (KarthikShanmugasundaram) [inline src/test/java/com/box/sdk/BoxCollaborationTest.java]: Extra line can be removed

File: src/test/java/com/box/sdk/BoxCollaborationTest.java [hunk 2]
```diff
@@ -50,4 +67,71 @@ public void deleteSucceeds() {
...
+            assertEquals(fileCollabInfo.getRole(), localFileCollabInfor.getRole());
+            assertEquals(fileCollabInfo.getStatus(), localFileCollabInfor.getStatus());
+        }
+
+        assertEquals(collabIterator.hasNext(), false);
+        assertEquals(2, numCollabs);
+
+        uploadedFile.delete();
+
+
```
Comment (KarthikShanmugasundaram) [inline src/test/java/com/box/sdk/BoxCollaborationTest.java]: Extra lines can be removed

File: src/main/java/com/box/sdk/BoxFolder.java [hunk 3]
```diff
@@ -165,6 +241,8 @@ public BoxSharedLink createSharedLink(BoxSharedLink.Access access, Date unshareD
         return collaborations;
     }
 
+
```
Comment (KarthikShanmugasundaram) [inline src/main/java/com/box/sdk/BoxFolder.java]: Extra line can be removed

File: src/main/java/com/box/sdk/BoxFile.java [hunk 4]
```diff
@@ -1112,4 +1162,78 @@ String toJSONValue() {
...
+            accessibleByField.add("type", "user");
+        } else if (collaborator instanceof BoxGroup) {
+            accessibleByField.add("type", "group");
+        } else {
+            throw new IllegalArgumentException("The given collaborator is of an unknown type.");
+        }
+
+        return this.collaborate(accessibleByField, role);
+    }
+
```
Comment (KarthikShanmugasundaram) [inline src/main/java/com/box/sdk/BoxFile.java]: Extra line can be removed

File: src/main/java/com/box/sdk/BoxCollaboration.java [hunk 5]
```diff
@@ -432,4 +451,32 @@ String toJSONString() {
...
+        return new BoxResourceIterable<BoxCollaboration.Info>(
+                api, GET_ALL_FILE_COLLABORATIONS_URL.buildWithQuery(api.getBaseURL(), builder.toString(), fileID),
+                pageSize) {
+
+            @Override
+            protected BoxCollaboration.Info factory(JsonObject jsonObject) {
+                String id = jsonObject.get("id").asString();
+                return new BoxCollaboration(api, id).new Info(jsonObject);
+            }
+
```
Comment (KarthikShanmugasundaram) [inline src/main/java/com/box/sdk/BoxCollaboration.java]: Extra line can be removed

**Pull request merged**